workspaces/mcp-chat/plugins/mcp-chat/src/components/RightPane/RightPane.tsx

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { FC } from 'react';
+import { useTheme } from '@mui/material/styles';
+import AddIcon from '@mui/icons-material/Add';
+import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import MemoryIcon from '@mui/icons-material/Memory';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import { ActiveMcpServers } from './ActiveMcpServers';
+import { ActiveTools } from './ActiveTools';
+import { ProviderStatus } from './ProviderStatus';
+import { BotIcon } from '../BotIcon';
+import { MCPServer } from '../../types';
+import { UseProviderStatusReturn, useAvailableTools } from '../../hooks';
+
+interface RightPaneProps {
+  sidebarCollapsed: boolean;
+  onToggleSidebar: () => void;
+  onNewChat: () => void;
+  mcpServers: MCPServer[];
+  onServerToggle: (serverName: string) => void;
+  providerStatus: UseProviderStatusReturn;
+}
+
+export const RightPane: FC<RightPaneProps> = ({
+  sidebarCollapsed,
+  onToggleSidebar,
+  onNewChat,
+  mcpServers,
+  onServerToggle,
+  providerStatus,
+}: RightPaneProps) => {
+  const theme = useTheme();
+  const { availableTools, isLoading: toolsLoading } =
+    useAvailableTools(mcpServers);
+
+  return (
+    <Box
+      sx={{
+        width: sidebarCollapsed ? 60 : 400,
```

Comment:
This is the reason why the sidebar animates also the content. I think it might be an option to replace the custom sidebar with a MUI Drawer. But your solution is working so no need to touch it now!

https://v5.mui.com/material-ui/react-drawer/#mini-variant-drawer